<a href="https://colab.research.google.com/github/susansunshin/Project-2/blob/main/Project_2.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#Project 2 - part 1

Susan Shin

1. Source of data

2. Brief description of data

3. What is the target?

4. What does one row represent? (A person?  A business?  An event? A product?)

5. Is this a classification or regression problem?

6. How many features does the data have?

7. How many rows are in the dataset?

8. What, if any, challenges do you foresee in cleaning, exploring, or modeling with this dataset?

##Data Set 1

In [5]:
filename = "/content/sample_data/train.csv"
df2 = pd.read_csv(filename)
df2

User_ID Product_ID Gender   Age  Occupation City_Category  \
0       1000001  P00069042      F  0-17          10             A   
1       1000001  P00248942      F  0-17          10             A   
2       1000001  P00087842      F  0-17          10             A   
3       1000001  P00085442      F  0-17          10             A   
4       1000002  P00285442      M   55+          16             C   
...         ...        ...    ...   ...         ...           ...   
203277  1001365  P00271142      F  0-17          10             C   
203278  1001365  P00251242      F  0-17          10             C   
203279  1001366  P00184942      M  0-17          10             C   
203280  1001366  P00237642      M  0-17          10             C   
203281  1001366  P00235242      M  0-17          10             C   

       Stay_In_Current_City_Years  Marital_Status  Product_Category_1  \
0                               2             0.0                 3.0   
1                               2             0.0                 1.0   
2                               2             0.0                12.0   
3                               2             0.0                12.0   
4                              4+             0.0                 8.0   
...                           ...             ...                 ...   
203277                          3             0.0                 5.0   
203278                          3             0.0                 5.0   
203279                          0             0.0                 1.0   
203280                          0             0.0                 5.0   
203281                          0             NaN                 NaN   

        Product_Category_2  Product_Category_3  Purchase  
0                      NaN                 NaN    8370.0  
1                      6.0                14.0   15200.0  
2                      NaN                 NaN    1422.0  
3                     14.0                 NaN    1057.0  
4                      NaN                 NaN    7969.0  
...                    ...                 ...       ...  
203277                 NaN                 NaN    5210.0  
203278                11.0                 NaN    5385.0  
203279                 8.0                17.0   15573.0  
203280                14.0                 NaN    5262.0  
203281                 NaN                 NaN       NaN  

[203282 rows x 12 columns]

1. Source of data 
https://www.kaggle.com/datasets/pranavuikey/black-friday-sales-eda

2. Brief description of data
Black Friday sales by customer by purchase

3. What is the target? Purchase

4. What does one row represent? (A person?  A business?  An event? A product?)
a customer transactioin
5. Is this a classification or regression problem? regression or classification

6. How many features does the data have? 12

7. How many rows are in the dataset? 203,282

8. What, if any, challenges do you foresee in cleaning, exploring, or modeling with this dataset? There is only 1 product category and I do not know what the numbers stand for. 

##Data Set 2

In [3]:
import pandas as pd
import numpy as np

from google.colab import drive
drive.mount('/content/drive')

filename = "/content/sample_data/Transactions.csv"
df = pd.read_csv(filename)
df

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


transaction_id  cust_id   tran_date  prod_subcat_code  prod_cat_code  \
0         80712190438   270351  28-02-2014                 1              1   
1         29258453508   270384  27-02-2014                 5              3   
2         51750724947   273420  24-02-2014                 6              5   
3         93274880719   271509  24-02-2014                11              6   
4         51750724947   273420  23-02-2014                 6              5   
...               ...      ...         ...               ...            ...   
23048     94340757522   274550  25-01-2011                12              5   
23049     89780862956   270022  25-01-2011                 4              1   
23050     85115299378   271020  25-01-2011                 2              6   
23051     72870271171   270911  25-01-2011                11              5   
23052     77960931771   271961  25-01-2011                11              5   

       Qty  Rate      Tax  total_amt Store_type  
0       -5  -772  405.300  -4265.300     e-Shop  
1       -5 -1497  785.925  -8270.925     e-Shop  
2       -2  -791  166.110  -1748.110   TeleShop  
3       -3 -1363  429.345  -4518.345     e-Shop  
4       -2  -791  166.110  -1748.110   TeleShop  
...    ...   ...      ...        ...        ...  
23048    1  1264  132.720   1396.720     e-Shop  
23049    1   677   71.085    748.085     e-Shop  
23050    4  1052  441.840   4649.840        MBR  
23051    3  1142  359.730   3785.730   TeleShop  
23052    1   447   46.935    493.935   TeleShop  

[23053 rows x 10 columns]

1. Source of data: https://www.kaggle.com/code/timidoe/retail-sales-data-analysis

2. Brief description of data : data is transaction by customer

3. What is the target? The target is the total_amount

4. What does one row represent? (A person?  A business?  An event? A product?) a sales transaction

5. Is this a classification or regression problem?
Regression

6. How many features does the data have?
10

7. How many rows are in the dataset?
23,052

8. What, if any, challenges do you foresee in cleaning, exploring, or modeling with this dataset?
I know that date is something that we shouldn't be using, but if I could understand how to incorporate, I would love to use this data set as it is actually applied to real life for me.